### PR TITLE
Fix bug with `BigIntArray` serialization

### PR DESCRIPTION
**Pull request by nik9000**
This fixed a bug with the `BigIntArray` serialization that I wrote in #89668 where it'd skip the entire final block if we used all of it.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/common/bytes/ReleasableBytesReferenceStreamInputTests.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 import static org.hamcrest.Matchers.equalTo;
 
-public class ReleasableBytesReferenceStreamTests extends AbstractStreamTests {
+public class ReleasableBytesReferenceStreamInputTests extends AbstractStreamTests {
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/common/bytes/ReleasableBytesReferenceStreamInputTests.java:28]: This lines the test name up with the rest of the subclasses.

File: server/src/test/java/org/elasticsearch/common/io/stream/AbstractStreamTests.java [hunk 2]
```diff
@@ -222,7 +222,11 @@ public void testSmallBigIntArray() throws IOException {
     }
 
     public void testLargeBigIntArray() throws IOException {
-        assertBigIntArray(between(PageCacheRecycler.INT_PAGE_SIZE, 10000));
+        assertBigIntArray(between(PageCacheRecycler.INT_PAGE_SIZE, 5_000_000));
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/common/io/stream/AbstractStreamTests.java:225]: I originally thought a bigger value would yield it. It didn't, but the bigger value doesn't take too long anyway.

File: server/src/main/java/org/elasticsearch/common/util/BigIntArray.java [hunk 3]
```diff
@@ -54,11 +54,17 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         int intSize = (int) size;
         out.writeVInt(intSize * 4);
+        int lastPageEnd = intSize % INT_PAGE_SIZE;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/common/util/BigIntArray.java:57]: I think you could fix this with a one liner like `int lastPageEnd = (intSize + 1) % INT_PAGE_SIZE - 1;` but I think that's harder to read that this.

File: server/src/main/java/org/elasticsearch/common/util/BigIntArray.java [hunk 4]
```diff
@@ -54,11 +54,17 @@ public void writeTo(StreamOutput out) throws IOException {
...
+        if (lastPageEnd == 0) {
+            for (int i = 0; i < pages.length; i++) {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/common/util/BigIntArray.java:59]: NIT: I guess you could also do `for (page : pages) {` here?

**Review by original-brownbear — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/common/util/BigIntArray.java (hunk 3 above)
Comment (not-napoleon) [reply server/src/main/java/org/elasticsearch/common/util/BigIntArray.java:57]: Nit: on the line above, we should use `Integer.BYTES`, not a hard-coded 4

**Review by not-napoleon — APPROVED** [review]
LGTM

**Pull request merged**